src/discussions/in-context-topics/TopicPostsView.test.jsx

```diff
@@ -0,0 +1,280 @@
+  }
+
+  test.each([
+    { parent: nonCoursewareTopics[0], topicType: 'NonCourseware' },
+    { parent: coursewareTopics[0].children[0].children[0], topicType: 'courseware' },
+  ])('\'$topicType\' topic should have a required number of post lengths.', async ({ parent }) => {
+    await setupTopicsMockResponse();
+
+    await act(async () => {
+      setupPostsMockResponse(parent.id, 3);
+      renderComponent({ topicId: parent.id });
+    });
+
+    await waitFor(async () => {
+      const posts = await container.querySelectorAll('.discussion-post');
+
+      expect(lastLocation.pathname.endsWith(`/topics/${parent.id}`)).toBeTruthy();
+      expect(posts).toHaveLength(6);
+    });
+  });
+
+  test.each([
+    { parent: nonCoursewareTopics[0], topicType: 'NonCourseware' },
+    { parent: coursewareTopics[0].children[0].children[0], topicType: 'courseware' },
+  ])('\'$topicType\' topic should have a Back navigation bar with selected \'$topicType\' topic title.',
+    async ({ parent }) => {
+      await setupTopicsMockResponse();
+
+      await act(async () => {
+        setupPostsMockResponse(parent.id, 3);
+        renderComponent({ topicId: parent.id });
+      });
+
+      await waitFor(async () => {
+        const backButton = await screen.getByLabelText('Back to topics list');
+        const parentHeader = await screen.findByText(parent.name);
+
+        expect(backButton).toBeInTheDocument();
+        expect(parentHeader).toBeInTheDocument();
+      });
+    });
+
+  it('A back button should redirect from list of posts to list of units.', async () => {
+    await setupTopicsMockResponse();
+    const subSection = coursewareTopics[0].children[0];
+    const unit = subSection.children[0];
+
+    await act(async () => {
+      setupPostsMockResponse(unit.id, 2);
+      renderComponent({ topicId: unit.id });
+    });
+
+    const backButton = await screen.getByLabelText('Back to topics list');
+
+    await act(async () => fireEvent.click(backButton));
+    await waitFor(async () => {
+      renderComponent({ topicId: undefined, category: subSection.id });
+
+      const subSectionList = await container.querySelector('.list-group');
+      const units = await subSectionList.querySelectorAll('.discussion-topic');
+      const unitHeader = await within(subSectionList).queryByText(unit.name);
+
+      expect(lastLocation.pathname.endsWith(`/category/${subSection.id}`)).toBeTruthy();
+      expect(unitHeader).toBeInTheDocument();
+      expect(units).toHaveLength(4);
+    });
+  });
+
+  it('A back button should redirect from units to the parent/selected subsection.', async () => {
+    await setupTopicsMockResponse();
+    const subSection = coursewareTopics[0].children[0];
+
+    renderComponent({ topicId: undefined, category: subSection.id });
+
+    const backButton = await screen.getByLabelText('Back to topics list');
+
+    await act(async () => fireEvent.click(backButton));
+    await waitFor(async () => {
+      renderComponent();
```

Comment:
Please check if the back button Url is generating or not. If It is generating then it should redirect to units pages. Do not render a component more than once in a test.

Reply:
i need to rerender component based on updated props value to rerender is necessary. Multiple rerender is acceptable.
https://testing-library.com/docs/example-update-props/